File: Vacas Claustrofóbicas/Vacas Claustrofóbicas.pas
program Hd;
uses math;

type
    point = record
        x, y: Int64;
    end;

var
    n, x, y, i, j: Int64;
    coord: array of point;
    m, x1, y1: real;

begin
    readln(n);
    SetLength(coord, n);
    
    for i:=0 to n-1 do
        readln(coord[i].x, coord[i].y);
    
    m := 100000000.0;
    
    for i:=0 to n-2 do
    begin
        for j:=i+1 to n-1 do
        begin
            x1 := coord[i].x - coord[j].x;
            x1 := x1 * x1;
            y1 := coord[i].y - coord[j].y;
            y1 := y1 * y1;
            y1 := sqrt(x1 + y1);
            
            if m > y1 then
            begin
                m := y1;
                x := i + 1;
                y := j + 1;
            end;
        end;
    end;
    writeln(x, ' ', y);
end.

File: Vacas Claustrofóbicas/Vacas Claustrofóbicas.c
#include<stdio.h>
#include<math.h>
#define Hd main(void)

struct point{
    int x;
    int y;
};

Hd
{
    int n;
    scanf("%d",&n);
    struct point coord[n];
    for(int i=0;i<n;i++){
        scanf("%d %d",&coord[i].x,&coord[i].y);
    }
    int x,y;
    float m=100000000.0;
    for(int i=0;i<n-1;i++){
        for(int j=i+1;j<n;j++){
            float x1=coord[i].x-coord[j].x;
            x1*=x1;
            float y1=coord[i].y-coord[j].y;
            y1*=y1;
            y1= sqrt(x1+y1);
            if(m>y1){
                m=y1;
                x=i+1;
                y=j+1;
            }
        }
    }
    printf("%d %d",x,y);
}